trackpy/packing.py

```diff
@@ -0,0 +1,130 @@
+from scipy.spatial import cKDTree
+import random
+import numpy as np
+from warnings import warn
+
+
+
+def pairCorrelation2D(feat, fraction = 1., dr = .5, cutoff = 20., ndensity=None, boundary = None):
+    """
+    Calculate the pair correlation function in 2 dimensions.
+
+    Parameters
+    ----------
+    feat : Pandas DataFrame
+        DataFrame containing the x and y coordinates of particles
+    fration : float, optional
+        The fraction of particles to calculate g(r) with. May be used to increase speed of function. Particles selected at random.
+    dr : float, optional
+        The bin width
+    cutoff : float, optional
+        Maximum distance to calculate g(r)
+    ndensity : float, optional
+        Density of particle packing. If not specified, density will be calculated assuming rectangular homogenous arangement
+    boundary : Tuple, optional
+        Tuple specifying rectangular boundary of partcicles (xmin, xmax, ymin, ymax)
+
+    Returns
+    -------
+    r_edges : array
+        Return the bin edges
+    g_r : array
+        The values of g_r
+    """
+
+    if boundary is None:
+        warn("Rectangular packing is assumed. Boundaries are determined by edge particles.")
+        xmin = feat.x.min()
+        xmax = feat.x.max()
+        ymin = feat.y.min()
+        ymax = feat.y.max()
+    else:
+        xmin = boundary[0]
+        xmax = boundary[1]
+        ymin = boundary[2]
+        ymax = boundary[3]
+
+    if ndensity is None:
+        ndensity = feat.x.count() / ((feat.x.max() - feat.x.min()) * (feat.y.max() - feat.y.min())) #particle packing density
+        warn("Rectangular homogenous packing is assumed to calculate particle density.")
+
+
+    p_indexes = random.sample(range(len(feat)), int(fraction*len(feat))) #grab random sample of particles
+    r_edges = np.arange(dr, cutoff + 2*dr, dr) #radii to search for particles
+    g_r = np.zeros(len(r_edges) - 1)
+    max_p_count =  int(np.pi * (r_edges.max() + dr)**2 * ndensity * 10) #upper bound for neighborhood particle count
+    ckdtree = cKDTree(feat[['x', 'y']])#initialize kdtree
```

Comment:
You should take a look at the PEP8 conventions. We follow them loosely; see the [information for contributors](https://github.com/soft-matter/trackpy/wiki/Information-for-contributors). One convention trackpy generally adheres to is to put two spaces before an inline comment, and one space between the "#" and the comment text.